rest/config_test.go

```diff
@@ -518,29 +518,98 @@ func TestMergeWith(t *testing.T) {
 
 func TestDeprecatedConfigLoggingFallback(t *testing.T) {
 	defer base.SetUpTestLogging(base.LevelInfo, base.KeyAll)()
-	logFilePath := "/var/log/sync_gateway"
-	dlfPath := "/tmp/log/sync_gateway"
 	logKeys := []string{"Admin", "Access", "Auth", "Bucket", "Cache"}
 	deprecatedLog := []string{"Admin", "Access", "Auth", "Bucket", "Cache"}
 
-	ddl := &base.LogAppenderConfig{LogFilePath: &logFilePath, LogKeys: logKeys, LogLevel: base.PanicLevel}
-	lc := &base.LoggingConfig{DeprecatedDefaultLog: ddl}
-	sc := &ServerConfig{Logging: lc, DeprecatedLogFilePath: &dlfPath, DeprecatedLog: deprecatedLog}
-
-	warns := sc.deprecatedConfigLoggingFallback()
-	assert.Equal(t, filepath.Dir(*sc.Logging.DeprecatedDefaultLog.LogFilePath), sc.Logging.LogFilePath)
-	assert.Equal(t, sc.Logging.DeprecatedDefaultLog.LogKeys, sc.Logging.Console.LogKeys)
-	assert.Equal(t, base.ToLogLevel(sc.Logging.DeprecatedDefaultLog.LogLevel), sc.Logging.Console.LogLevel)
-	assert.Equal(t, sc.DeprecatedLog, sc.Logging.Console.LogKeys)
-	assert.Len(t, warns, 3)
-
-	// Call deprecatedConfigLoggingFallback without DeprecatedDefaultLog
-	sc = &ServerConfig{Logging: &base.LoggingConfig{}, DeprecatedLogFilePath: &dlfPath, DeprecatedLog: deprecatedLog}
-	warns = sc.deprecatedConfigLoggingFallback()
-
-	assert.Equal(t, *sc.DeprecatedLogFilePath, sc.Logging.LogFilePath)
-	assert.Equal(t, sc.DeprecatedLog, sc.Logging.Console.LogKeys)
-	assert.Len(t, warns, 2)
+	removeDirs := func(dirs ...string) {
+		for _, dir := range dirs {
+			require.NoErrorf(t, os.RemoveAll(dir), "Error removing dir %q", dir)
+			_, err := os.Stat(dir)
+			require.True(t, os.IsNotExist(err), "Removed dir %q shouldn't exist", dir)
+		}
+	}
+
```

Comment:
As discussed, subtests have a lot more value if they are defined up front, and then t.Run is just executing the same functionality against different input conditions, and verifying against the different expected results.

Reply:
This test has been refactored by using table tests.